### PR TITLE
Allows trace scope decoration such as Log4J trace ID correlation

### PR DESCRIPTION
**Pull request by adriancole**
This allows us to do things now that we are using a custom thread local.

See https://github.com/openzipkin/brave/pull/683

### CODE REVIEW

File: zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java [hunk 1]
```diff
@@ -50,9 +50,11 @@
      * Use this singleton when building a {@link brave.Tracing} instance for use with
      * {@link HttpTracingService} or {@link HttpTracingClient}.
      *
+     * <p>If you need to customize the context, use {@linkplain #newBuilder()} instead.
+     *
      * @see brave.Tracing.Builder#currentTraceContext(brave.propagation.CurrentTraceContext)
      */
-    public static final CurrentTraceContext INSTANCE = new RequestContextCurrentTraceContext();
+    public static final CurrentTraceContext INSTANCE = new RequestContextCurrentTraceContext(new Builder());
```
Comment (trustin) [inline zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: How about renaming to `DEFAULT`?

Comment (adriancole) [reply zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: INSTANCE has been released.. should we still do this?

**Comment by adriancole** [comment]
ok corresponding brave release is out

File: zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java (hunk 1 above)
Comment (trustin) [reply zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: That's a good question. How about deprecating `INSTANCE` and adding `DEFAULT`?

File: zipkin/src/test/java/com/linecorp/armeria/it/tracing/HttpTracingIntegrationTest.java [hunk 2]
```diff
@@ -196,8 +198,12 @@ private static HttpTracingService decorate(String name, Service<HttpRequest, Htt
     }
 
     private static Tracing newTracing(String name) {
+        CurrentTraceContext currentTraceContext =
```
Comment (trustin) [inline zipkin/src/test/java/com/linecorp/armeria/it/tracing/HttpTracingIntegrationTest.java]: nit: missing `final`

File: zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java (hunk 1 above)
Comment (adriancole) [reply zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: works for me

File: zipkin/src/test/java/com/linecorp/armeria/it/tracing/HttpTracingIntegrationTest.java (hunk 2 above)
Comment (adriancole) [reply zipkin/src/test/java/com/linecorp/armeria/it/tracing/HttpTracingIntegrationTest.java]: done

File: zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java [hunk 3]
```diff
@@ -50,9 +50,20 @@
...
-    public static final CurrentTraceContext INSTANCE = new RequestContextCurrentTraceContext();
+    public static final CurrentTraceContext DEFAULT = new RequestContextCurrentTraceContext(new Builder());
+
+    /**
+     * Singleton retained for backwards compatibility, but replaced by {@link #DEFAULT}.
+     *
+     * @deprecated Please use {@link #DEFAULT} or {@link #newBuilder()} to customize an instance.
+     */
+    @Deprecated
+    public static final CurrentTraceContext INSTANCE = new RequestContextCurrentTraceContext(new Builder());
```
Comment (trustin) [inline zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: Could be just `= DEFAULT`?

**Comment by adriancole** [comment]
you mean delete the INSTANCE one? yeah I think it would require few
apologies. say the word and it will be so

File: zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java (hunk 3 above)
Comment (adriancole) [reply zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: oh I understand :P

Comment (adriancole) [reply zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: disregard my other comment.. done!

File: zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java [hunk 4]
```diff
@@ -80,6 +91,27 @@ public String toString() {
...
+            return new RequestContextCurrentTraceContext(this);
+        }
+
+        Builder() {
+        }
+    }
+
+    /**
+     * Use this when you need customizations such as log integration via
+     * {@linkplain Builder#addScopeDecorator(ScopeDecorator)}.
```
Comment (minwoox) [inline zipkin/src/main/java/com/linecorp/armeria/common/tracing/RequestContextCurrentTraceContext.java]: nit: `@linkplain` -> `@link`

**Comment by trustin** [comment]
Thanks, @adriancole !

**Pull request merged**